File: app/src/main/java/co/tpcreative/supersafe/model/DriveDescription.java
package co.tpcreative.supersafe.model;
import com.google.gson.Gson;
import java.io.Serializable;
import co.tpcreative.supersafe.common.util.Utils;

public class DriveDescription implements Serializable{

    private static String TAG = DriveDescription.class.getSimpleName();
    public String categories_local_id;
    public String categories_id;
    public String fileExtension;
    public String thumbnailPath;
    public String originalPath;
    public String title;
    public String global_original_id;
    public String global_thumbnail_id;
    public String mimeType;
    public String thumbnailName;
    public String originalName;
    public String items_id;
    public int degrees;
    public int formatType;
    public boolean thumbnailSync;
    public boolean originalSync;
    public int fileType;
    public String size;
    public boolean isDeleteLocal;
    public boolean isDeleteGlobal;



    public int statusProgress;
    public int deleteAction;
    public boolean isFakePin;
    public boolean isSaver;
    public boolean isExport;
    public boolean isWaitingForExporting;
    public int custom_items;

    private static DriveDescription instance;

    public static DriveDescription getInstance(){
        if (instance==null){
            instance = new DriveDescription();
        }
        return instance;
    }

    public String convertToHex(final String value){
        try {
            return Utils.stringToHex(value);
        }
        catch (Exception e){
            Utils.Log(TAG,"Error :" + e.getMessage());
            e.printStackTrace();
            Utils.onWriteLog(TAG +"-" + e.getMessage(),EnumStatus.WRITE_FILE);
        }
        return null;
    }

    public DriveDescription hexToObject(String value){
        try {
            String result = Utils.hexToString(value);
            return new Gson().fromJson(result,DriveDescription.class);
        }
        catch (Exception e){
            Utils.Log(TAG,"Error :" + e.getMessage());
            e.printStackTrace();
        }
        return null;
    }

}

File: app/src/main/java/co/tpcreative/supersafe/model/DriveDescription.kt
package co.tpcreative.supersafe.model
import co.tpcreative.supersafe.common.util.Utils
import com.google.gson.Gson
import java.io.Serializable

class DriveDescription : Serializable {
    var categories_local_id: String? = null
    var categories_id: String? = null
    var fileExtension: String? = null
    var thumbnailPath: String? = null
    var originalPath: String? = null
    var title: String? = null
    var global_original_id: String? = null
    var global_thumbnail_id: String? = null
    var mimeType: String? = null
    var thumbnailName: String? = null
    var originalName: String? = null
    var items_id: String? = null
    var degrees = 0
    var formatType = 0
    var thumbnailSync = false
    var originalSync = false
    var fileType = 0
    var size: String? = null
    var isDeleteLocal = false
    var isDeleteGlobal = false
    var statusProgress = 0
    var deleteAction = 0
    var isFakePin = false
    var isSaver = false
    var isExport = false
    var isWaitingForExporting = false
    var custom_items = 0
    fun convertToHex(value: String?): String? {
        try {
            return Utils.stringToHex(value)
        } catch (e: Exception) {
            Utils.Log(TAG, "Error :" + e.message)
            e.printStackTrace()
            Utils.onWriteLog(TAG + "-" + e.message, EnumStatus.WRITE_FILE)
        }
        return null
    }

    fun hexToObject(value: String?): DriveDescription? {
        try {
            val result: String? = Utils.hexToString(value)
            return Gson().fromJson(result, DriveDescription::class.java)
        } catch (e: Exception) {
            Utils.Log(TAG, "Error :" + e.message)
            e.printStackTrace()
        }
        return null
    }

    companion object {
        private val TAG = DriveDescription::class.java.simpleName
        private var instance: DriveDescription? = null
        fun getInstance(): DriveDescription? {
            if (instance == null) {
                instance = DriveDescription()
            }
            return instance
        }
    }
}